beacon-chain/sync/subscriber.go

```diff
@@ -227,14 +262,29 @@ func (s *Service) subscribeStaticWithSubnets(topic string, validator pubsub.Vali
 				if s.chainStarted.IsSet() && s.cfg.InitialSync.Syncing() {
 					continue
 				}
+				valid, err := isDigestValid(digest, genesis, genRoot)
+				if err != nil {
+					log.Error(err)
+					continue
+				}
+				if !valid {
+					log.Warnf("Attestation subnets with digest %#x are no longer valid, unsubscribing from all of them.", digest)
+					// Unsubscribes from all our current subnets.
+					for i := uint64(0); i < params.BeaconNetworkConfig().AttestationSubnetCount; i++ {
+						fullTopic := fmt.Sprintf(topic, digest, i) + s.cfg.P2P.Encoding().ProtocolSuffix()
```

Comment:
This can be abstracted as well as part of the abstraction of topic contains functions. We should have a topic builder abstraction that can be done in a separate cleanup